site/app/controllers/OfficeHoursQueueController.php

```diff
@@ -32,31 +31,58 @@ public function showQueue() {
                 new RedirectResponse($this->core->buildCourseUrl(['home']))
             );
         }
-        if (!$this->core->getUser()->accessGrading()) {
-            $oh_queue = $this->core->getQueries()->getQueueByUser($this->core->getUser()->getId());
-            return Response::WebOnlyResponse(
-                new WebResponse(
-                    'OfficeHoursQueue',
-                    'showQueueStudent',
-                    $oh_queue,
-                    $this->core->getConfig()->getCourse()
-                )
+
+        return Response::WebOnlyResponse(
+            new WebResponse(
+                'OfficeHoursQueue',                      //Goes to this file OfficeHoursQueueView.php
+                'showTheQueue',                          //Runs this functin showTheQueue()
+                new OfficeHoursQueueViewer($this->core)  //Passing in this variable which is a OfficeHoursQueueViewer object
+            )
+        );
+    }
+
+    /**
+    * @Route("/{_semester}/{_course}/office_hours_queue/open", methods={"POST"})
```

Comment:
For many of these routes, let's add the `queue_code` as part of the route, as it's the direct resource that we're acting upon (which the route represents).

So, this one might be `/{_semester}/{_course}/office_hours_queue/{queue_code}/open`, etc.

The big change would be `/{_semester}/{_course}/office_hours_queue/new`, and I would just drop the `/new` bit and make POSTs to `/office_hour_queue` represent creation of new queues.